### PR TITLE
Feature/warn cli UI secrets backend

### LINKED ISSUE #10867: Warning about invisible connections/variable in the Web UI
Hello,

Currently, you can define a new connection/variable in several ways.
* Metabase
* Environment variable
* Secret backend

However, only the connections defined with the Metabase are visible from the Web UI. This is surprising among users because a natural way to check connection/variable configuration is to use the Web UI, which may not show complete information.
I would be happy if there was a warning stating that not all connections / variables are visible when the user has a configured secret backend or when the `AIRFLOW__CONN_*` /`AIRFLOW__VAR_*`  environment variable is defined. This should appear above the list of connections/variables.

This change is intended to improve the user experience.

Best regards,
Kamil Breguła

### PR DESCRIPTION
**Pull request by RNHTTR**
I accidentally closed #24679 when I moved the code to a new branch in my fork, and I can't seem to be able to point the existing PR to the new branch & re-open the PR. So, I just opened this PR instead.

---
The Airflow CLI and UI don't show Variables or Connections that are stored in a secrets backend. This can be really confusing if you don't already know that. This PR adds a warning to CLI commands and a warning banner to the list connections and list variables views in the UI when a user has enabled a secrets backend. The warning just explains that connections and variables stored in secrets backends won't be displayed using the CLI or UI.

Closes: https://github.com/apache/airflow/issues/10867

### CODE REVIEW

**Comment by o-nikolas** [comment]
CC: @blag @Taragolis

**Comment by potiuk** [comment]
It's cool and it will remove a source of confusion for our users. 

First of all - you will need to rebase @RNHTTR - there are some conflicts.

One general comment I have - the warning message mentions EnvironmentVariablesBackend but this one is not really part of custom secret backends so the use of it is not really detected. 

I thought we should really treat Environment variables a little differently than other custom secrets. As opposed to all other secrets we can actually check if any custom connection/variable is defined in Environment and print the warning when it is. We just have to check if there are anny AIRFLOW__CONNECTION__ env variables or AIRFLOW__VARIABLES__.* 

If you do it this way (additionally to checking if there is a custom backend defined), the warning will **also** be printed when  there are any custom variables defined via env variables (and you can slightly modify the warning message to explain it).

I think that would be a more comple solution to the problem.

File: airflow/cli/commands/connection_command.py [hunk 1]
```diff
@@ -75,13 +79,20 @@ def connections_get(args):
 @suppress_logs_and_warning
 def connections_list(args):
     """Lists all connections at the command line"""
+    airflow_console = AirflowConsole()
+    if CUSTOM_BACKEND:
+        warning = warn_list_secrets_alternative_backend(cli_or_ui="cli", connection_or_variable="connection")
+        airflow_console.print(
+            f'WARNING: {warning}',
+            style="magenta",
+        )
```
Comment (uranusjr) [inline airflow/cli/commands/connection_command.py]:
```suggestion
        airflow_console.print(f"WARNING: {warning}", style="magenta")
```

File: airflow/cli/commands/variable_command.py [hunk 2]
```diff
@@ -23,18 +23,29 @@
...
 
 from airflow.cli.simple_table import AirflowConsole
+from airflow.configuration import get_custom_secret_backend
 from airflow.models import Variable
 from airflow.utils import cli as cli_utils
 from airflow.utils.cli import suppress_logs_and_warning
 from airflow.utils.session import create_session
+from airflow.utils.warnings import warn_list_secrets_alternative_backend
+
+CUSTOM_BACKEND: bool = True if get_custom_secret_backend() else False
```
Comment (uranusjr) [inline airflow/cli/commands/variable_command.py:33]: This shouldn’t need to be repeated. Should we instead extract this to somewhere shared between both modules?

File: airflow/cli/commands/variable_command.py [hunk 3]
```diff
@@ -23,18 +23,29 @@
...
 @suppress_logs_and_warning
 def variables_list(args):
     """Displays all of the variables"""
+    airflow_console = AirflowConsole()
+    if CUSTOM_BACKEND:
+        warning = warn_list_secrets_alternative_backend(cli_or_ui="cli", connection_or_variable="variable")
+        airflow_console.print(
+            f'WARNING: {warning}',
+            style="magenta",
+        )
```
Comment (uranusjr) [inline airflow/cli/commands/variable_command.py]:
```suggestion
        airflow_console.print(f"WARNING: {warning}", style="magenta")
```

**Review by uranusjr — CHANGES_REQUESTED** [review]
Also need to update against the newest changes in main.

**Comment by potiuk** [comment]
Static checks ?

**Comment by potiuk** [comment]
Still working on it @RNHTTR  ? needs rebase and fixes :(

**Pull request closed without merging**